extension/app/ts/background/iconHandler.ts

```diff
@@ -67,6 +67,20 @@ export async function updateExtensionBadge() {
 }
 
 export async function retrieveWebsiteDetails(port: browser.runtime.Port, websiteOrigin: string) {
+	const tryGettingTab = async (tabId: number) => {
+		try {
+			return browser.tabs.get(tabId)
+		} catch (error) {
+			if (error instanceof Error) {
+				if (error.message?.includes('No tab with id')) {
+					// if tab is not found (user might have closed it)
+					return undefined
+				}
+			}
+			throw error
```

Comment:
Consider inverting the conditional with guard clauses.
```suggestion
			if (!(error instanceof Error)) throw error
			if (!error.message?.includes('No tab with id')) throw error
			// if tab is not found (user might have closed it)
			return undefined
```